### PR TITLE
Add Thoth Service Metrics dashboard

### PR DESCRIPTION
**Pull request by pacospace**
Related-To: https://github.com/thoth-station/thoth-application/pull/826

### CODE REVIEW

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml
Comment (tumido) [inline odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: I think (?) Grafana requires this name to contain the file type suffix (`.json`), because it translates to a filename which is then mounted to Grafana:

https://github.com/operate-first/apps/blob/master/docs/odh/grafana/add_grafana_dashboard.md#add-a-grafana-dashboard-to-operatefirst-grafana
https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md

**Review by tumido — CHANGES_REQUESTED** [review]

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml
Comment (pacospace) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: Thanks @tumido, then I have to update all other dashboards as well. see for example: https://github.com/operate-first/apps/blob/7fac10d12109b0b53d9750bdb9c6269ce3bcc0d9/odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-knowledge-graph-content-metrics.yaml#L9 and all others Thoth related: https://github.com/operate-first/apps/tree/master/odh/overlays/moc/monitoring/dashboards/thoth-station cc @harshad16

Comment (tumido) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: Oh, does work just fine for the other dashboards without the suffix? Maybe I interpret it wrong and you can leave it.. :smile:

Comment (pacospace) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: Even though it appears weird to have requirement to put `.json` in the name, dashboards can be shared in other formats in Grafana: https://github.com/integr8ly/grafana-operator/blob/faceedabf2864402cc45767fe44ccd56e1b7955a/pkg/apis/integreatly/v1alpha1/grafanadashboard_types.go#L20

Comment (pacospace) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: Maybe we need to improve doc upstream to grafana operator and operate first doc consequently :D

Comment (pacospace) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: https://github.com/integr8ly/grafana-operator/issues/339

Comment (HumairAK) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:9]: Seems like this request is no longer needed, @tumido  can you resolve it?

**Comment by HumairAK** [comment]
The kustomize build on this directory fails: 
```
Error: accumulating resources: 2 errors occurred:
        * accumulateFile error: "merging resources from 'thoth-user-api-metrics.yaml': may not add resource with an already registered id: integreatly.org_v1alpha1_GrafanaDashboard|~X|thoth-user-api-metrics"
        * loader.New error: "error loading thoth-user-api-metrics.yaml with git: url lacks orgRepo: thoth-user-api-metrics.yaml, dir: got 
```
This is because `thoth-user-api-metrics.yaml` has the same resource kind/name.

**Review by tumido — APPROVED** [review]
:+1: I've been told to stop complaining about everything.. So be it. :smile:

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml
Comment (tumido) [inline odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:4]: Name doesn't match the file. I think you forgot to change this

**Review by tumido — CHANGES_REQUESTED** [review]

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml
Comment (tumido) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-service-metrics.yaml:4]: Ah.. yes... @HumairAK is faster again.. https://github.com/operate-first/apps/pull/191#issuecomment-770904582

**Comment by pacospace** [comment]
> The kustomize build on this directory fails:
> 
> ```
> Error: accumulating resources: 2 errors occurred:
>         * accumulateFile error: "merging resources from 'thoth-user-api-metrics.yaml': may not add resource with an already registered id: integreatly.org_v1alpha1_GrafanaDashboard|~X|thoth-user-api-metrics"
>         * loader.New error: "error loading thoth-user-api-metrics.yaml with git: url lacks orgRepo: thoth-user-api-metrics.yaml, dir: got 
> ```
> 
> This is because `thoth-user-api-metrics.yaml` has the same resource kind/name.

Yes thank you :) I forgot to send new commit, as I was waiting from grafana operator people to know about: https://github.com/integr8ly/grafana-operator/issues/339

**Comment by anishasthana** [comment]
Could you squash the commits @pacospace  :-D

**Comment by pacospace** [comment]
> Could you squash the commits @pacospace :-D

Why @anishasthana ? :D

**Comment by anishasthana** [comment]
Only the first commit out of the three has a message that tells me anything useful. Since all three commits are for the same feature, it'd make sense for them to be one commit. 
image

**Comment by pacospace** [comment]
> Only the first commit out of the three tells me anything useful. Since all three commits are for the same feature, it'd make sense for them to be one commit.
> image

The feature is in the PR title for me, which can be made by different commits (you don't have to add only a dashboard, but link it in kustomize and also you have to update all data not just link in this case) :) Moreover, different changes might come in a PR based on comments, I prefer to see the commit changes in that case. Moreover for others to learn and to easily see what changes have been updated, but that is personal I think :)

**Comment by anishasthana** [comment]
Re: different changes based in a PR based on comments, I'm a fan of [fixup](https://fle.github.io/git-tip-keep-your-branch-clean-with-fixup-and-autosquash.html) for PR reviews. I'm probably being overly pedantic here though, I like keeping the commit history clean in general.  Since I'm not an approver here though I'm not going to block on it :-)

**Comment by HumairAK** [comment]
I agree with @anishasthana here. In my opinion the pros of squashing commits outweighs the cons. This is a single change, so it should be reflected in a single commit "Add thoth service metrics dashboard". This makes debugging easier/faster if we're trying to find the source of a change that introduced a bug. It keeps the history clean without introducing bloated commits like "update previous commit by fixing a typo". It also makes it easier to handle merge conflicts (since you're dealing with a lot fewer conflicting commits).

**Review by harshad16 — COMMENTED** [review]
/lgtm
thanks for this addition

**Comment by harshad16** [comment]
I believe squashing the commits in this PR would be good, as this pr is focusing on a particular dashboard setup.
one commit with all the information is good as this is part of one feature addition. 
multiple commits make sense when each additional commit is corresponding to a different feature.

**Comment by pacospace** [comment]
Thanks @harshad16 @anishasthana @HumairAK I won't argue more in this case as you are the maintainers here. So I follow your change requested in this case. 

Can you please provide a contributor doc specifying this for this project? Because as I mention it is a personal decision and we could discuss for some time about pros/cons but in this case, I'm a contributor here so I ll squash them, but maybe a PR template specyfing this might help others from the beginning :) Thanks!

**Review by tumido — APPROVED** [review]

**Pull request merged**
